### DISCUSSION TITLE
Proxmox NGINX trouble

### DISCUSSION THREAD

**Discussion by K3vb3rt** [tteck/Proxmox#3460] [Help Needed] [answered]
Hi together
I installed NGINX using the script and then configured it successfully. After rebooting the machine, the container also starts successfully, but the web interface is not accessible. A PING to the IP is successful.

Could this have something to do with it?

root@nginxproxymanager:~# systemctl --failed
  UNIT                    LOAD   ACTIVE SUB    DESCRIPTION                       
● sys-kernel-config.mount loaded failed failed Kernel Configuration File System  
● openresty.service       loaded failed failed The OpenResty Application Platform
2 loaded units listed.

I really need help here, I don't want to completely reconfigure the container.

Thanks in advance!
Kev

**Comment by Mellowlynx** [comment]
Check `systemctl status [service name]`
To see the last log lines and check what was happening.
If you made changes in the config of these service, revered them back to default to see if it works then.

**Reply by K3vb3rt** [reply]
Thank you, but I can see that the service is running. 

Active: active (running) since Sat 2024-08-03 10:44:06 CEST; 9h ago

But I can't reach the web interface.

But there is this:

Aug 03 16:44:08 nginxproxymanager node[133]: [8/3/2024] [4:44:08 PM] [IP Ranges] › ✖  error     nginx: [warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in /data>
Aug 03 16:44:08 nginxproxymanager node[133]: nginx: [warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in /data/nginx/proxy_host/4.conf:20
Aug 03 16:44:08 nginxproxymanager node[133]: nginx: [error] invalid PID number "" in "/usr/local/openresty/nginx/logs/nginx.pid"

**Reply by Mellowlynx** [reply]
I have them too, so that's not the problem.
openresty should be running, that is needed for the UI, did you also check why that one? What's the error?

**Reply by K3vb3rt** [reply]
Sorry, didn't know that this also works with the service. Here is the log :-)

root@nginxproxymanager:~# systemctl status openresty
× openresty.service - The OpenResty Application Platform
     Loaded: loaded (/lib/systemd/system/openresty.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Sat 2024-08-03 10:44:10 CEST; 10h ago
    Process: 224 ExecStartPre=/usr/local/openresty/nginx/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
    Process: 241 ExecStart=/usr/local/openresty/nginx/sbin/nginx -g daemon on; master_process on; (code=exited, status=1/FAILURE)
        CPU: 31ms

Aug 03 10:44:08 nginxproxymanager nginx[241]: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
Aug 03 10:44:08 nginxproxymanager nginx[241]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Aug 03 10:44:09 nginxproxymanager nginx[241]: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
Aug 03 10:44:09 nginxproxymanager nginx[241]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Aug 03 10:44:09 nginxproxymanager nginx[241]: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
Aug 03 10:44:09 nginxproxymanager nginx[241]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Aug 03 10:44:10 nginxproxymanager nginx[241]: nginx: [emerg] still could not bind()
Aug 03 10:44:10 nginxproxymanager systemd[1]: openresty.service: Control process exited, code=exited, status=1/FAILURE
Aug 03 10:44:10 nginxproxymanager systemd[1]: openresty.service: Failed with result 'exit-code'.
Aug 03 10:44:10 nginxproxymanager systemd[1]: Failed to start openresty.service - The OpenResty Application Platform.

**Reply by Mellowlynx** [reply, marked as answer]
Did you change configs?

```
Aug 03 10:44:08 nginxproxymanager nginx[241]: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
Aug 03 10:44:08 nginxproxymanager nginx[241]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Aug 03 10:44:09 nginxproxymanager nginx[241]: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
Aug 03 10:44:09 nginxproxymanager nginx[241]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Aug 03 10:44:09 nginxproxymanager nginx[241]: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
Aug 03 10:44:09 nginxproxymanager nginx[241]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Aug 03 10:44:10 nginxproxymanager nginx[241]: nginx: [emerg] still could not bind()
Aug 03 10:44:10 nginxproxymanager systemd[1]: openresty.service: Control process exited, code=exited, status=1/FAILURE
Aug 03 10:44:10 nginxproxymanager systemd[1]: openresty.service: Failed with result 'exit-code'.
Aug 03 10:44:10 nginxproxymanager systemd[1]: Failed to start openresty.service - The OpenResty Application Platform.
```
Shows it can't bind to port 80 to start accepting incoming connections.
Something you change of installed is using that port.

Revered you changed back, so maybe it will work again.

**Reply by K3vb3rt** [reply]
Because I haven't made any changes, and I don't know what else is listening on port 80, I have now set up the container again. 
Now it is running again. Thank you!

**Reply by cal3k** [reply]
Having the same issue all of a sudden... Just noticed it after a proxmox reboot, not sure how long it's been going on since none of my backups (30 days) restores that functionality.
I'm using same vms' same setup since forever 

```
root@ngx:~# systemctl status openresty
x openresty.service - The OpenResty Application Platform
     Loaded: loaded (/lib/systemd/system/openresty.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Sat 2024-08-24 15:12:24 BST; 1min 57s ago
    Process: 149 ExecStartPre=/usr/local/openresty/nginx/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)
        CPU: 28ms

Aug 24 15:12:24 ngx nginx[149]: nginx: [warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in /data/nginx/proxy_host/7.conf:15
Aug 24 15:12:24 ngx nginx[149]: nginx: [warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in /data/nginx/proxy_host/8.conf:19
Aug 24 15:12:24 ngx nginx[149]: nginx: [warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in /data/nginx/proxy_host/8.conf:20
Aug 24 15:12:24 ngx nginx[149]: nginx: [warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in /data/nginx/proxy_host/9.conf:14
Aug 24 15:12:24 ngx nginx[149]: nginx: [warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in /data/nginx/proxy_host/9.conf:15
Aug 24 15:12:24 ngx nginx[149]: nginx: [emerg] socket() [::]:80 failed (97: Address family not supported by protocol)
Aug 24 15:12:24 ngx nginx[149]: nginx: configuration file /usr/local/openresty/nginx/conf/nginx.conf test failed
Aug 24 15:12:24 ngx systemd[1]: openresty.service: Control process exited, code=exited, status=1/FAILURE
Aug 24 15:12:24 ngx systemd[1]: openresty.service: Failed with result 'exit-code'.
```
Aug 24 15:12:24 ngx systemd[1]: Failed to start openresty.service - The OpenResty Application Platform.

And no, there does not appear to be something else listening on that port:

```
root@ngx:~# sudo netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      5081/sshd: /usr/sbi
tcp        0      0 0.0.0.0:3000            0.0.0.0:*               LISTEN      132/node
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      4817/master


```

**Comment by cal3k** [comment]
Some progress, I still can't access NPM web UI but, at least, now I know it does manage to start.

After removing openresty, I could reinstall it 

`dpkg --get-selections | grep openresty`

then clean the results

```
apt purge openresty -y
apt purge openresty-openssl111 -y
apt purge openresty-opm -y
apt purge openresty-zlib -y
apt autoremove -y
```
then reinstall it 

`apt install openresty -y`


Now I'm getting this
```
root@ngx:~#  netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      300/master
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1/init
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      161/nginx: master p
tcp        0      0 0.0.0.0:3000            0.0.0.0:*               LISTEN      131/node
```

```
root@ngx:~# systemctl status openresty
* openresty.service - The OpenResty Application Platform
     Loaded: loaded (/lib/systemd/system/openresty.service; enabled; preset: enabled)
     Active: active (running) since Sat 2024-08-24 22:37:14 BST; 12min ago
    Process: 146 ExecStartPre=/usr/local/openresty/nginx/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
    Process: 152 ExecStart=/usr/local/openresty/nginx/sbin/nginx -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
   Main PID: 161 (nginx)
      Tasks: 2 (limit: 77033)
     Memory: 8.9M
        CPU: 44ms
     CGroup: /system.slice/openresty.service
             |-161 "nginx: master process /usr/local/openresty/nginx/sbin/nginx -g daemon on; master_process on;"
             `-162 "nginx: worker process"

Aug 24 22:37:14 ngx systemd[1]: Starting openresty.service - The OpenResty Application Platform...
Aug 24 22:37:14 ngx systemd[1]: openresty.service: Failed to parse PID from file /usr/local/openresty/nginx/logs/nginx.pid: Invalid argument
Aug 24 22:37:14 ngx systemd[1]: Started openresty.service - The OpenResty Application Platform.

```
